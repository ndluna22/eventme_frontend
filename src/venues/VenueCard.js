import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import "./VenueCard.css";
import AspectRatio from "@mui/joy/AspectRatio";
import Button from "@mui/joy/Button";
import Card from "@mui/joy/Card";
import CardContent from "@mui/joy/CardContent";
import Typography from "@mui/joy/Typography";

function VenueCard({ id, name, images, city, state }) {
  console.debug("VenueCard", name);

  return (
    <Card sx={{ width: 300 }}>
      <div>
        <Typography level="title-lg">{name}</Typography>
      </div>
      <AspectRatio minHeight="120px" maxHeight="200px">
        <img src={images} srcSet={images} loading="lazy" alt="" />
      </AspectRatio>
      <CardContent orientation="horizontal">
        <div>
          <Typography level="body-xs">Location:</Typography>
          <Typography fontSize="lg" fontWeight="lg">
            {city}, {state}
          </Typography>
        </div>

        <Link to={`/venues/${id}`}>
          <Button
            variant="solid"
            size="md"
            color="primary"
            aria-label="Explore Venue"
            sx={{ ml: "auto", alignSelf: "center", fontWeight: 600 }}
          >
            Explore
          </Button>
        </Link>
      </CardContent>
    </Card>
  );
}

export default VenueCard;
