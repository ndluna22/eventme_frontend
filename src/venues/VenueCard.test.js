import React from "react";
import { render } from "@testing-library/react";
import VenueCard from "./VenueCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with name", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <VenueCard
        id="1234"
        name="Sphere"
        images="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without name", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <VenueCard
        id="5213"
        name="Sphere"
        images="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
