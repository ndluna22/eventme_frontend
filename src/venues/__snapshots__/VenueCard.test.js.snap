// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with name 1`] = `
<DocumentFragment>
  <div
    class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-ij021v-JoyCard-root"
  >
    <div
      data-first-child=""
    >
      <p
        class="MuiTypography-root MuiTypography-title-lg css-ar9ysw-JoyTypography-root"
      >
        Sphere
      </p>
    </div>
    <div
      class="MuiAspectRatio-root css-lqq0j9-JoyAspectRatio-root"
    >
      <div
        class="MuiAspectRatio-content MuiAspectRatio-variantSoft MuiAspectRatio-colorNeutral css-6e7lq1-JoyAspectRatio-content"
      >
        <img
          alt=""
          data-first-child=""
          loading="lazy"
          src="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
          srcset="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
        />
      </div>
    </div>
    <div
      class="MuiCardContent-root css-vpmu79-JoyCardContent-root"
      data-last-child=""
    >
      <div>
        <span
          class="MuiTypography-root MuiTypography-body-xs css-h7h2qx-JoyTypography-root"
        >
          Location:
        </span>
        <p
          class="MuiTypography-root MuiTypography-body-md css-1ygy07m-JoyTypography-root"
        >
          , 
        </p>
      </div>
      <a
        href="/venues/1234"
      >
        <button
          aria-label="Explore Venue"
          class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-cb19rj-JoyButton-root"
          type="button"
        >
          Explore
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot without name 1`] = `
<DocumentFragment>
  <div
    class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-ij021v-JoyCard-root"
  >
    <div
      data-first-child=""
    >
      <p
        class="MuiTypography-root MuiTypography-title-lg css-ar9ysw-JoyTypography-root"
      >
        Sphere
      </p>
    </div>
    <div
      class="MuiAspectRatio-root css-lqq0j9-JoyAspectRatio-root"
    >
      <div
        class="MuiAspectRatio-content MuiAspectRatio-variantSoft MuiAspectRatio-colorNeutral css-6e7lq1-JoyAspectRatio-content"
      >
        <img
          alt=""
          data-first-child=""
          loading="lazy"
          src="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
          srcset="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
        />
      </div>
    </div>
    <div
      class="MuiCardContent-root css-vpmu79-JoyCardContent-root"
      data-last-child=""
    >
      <div>
        <span
          class="MuiTypography-root MuiTypography-body-xs css-h7h2qx-JoyTypography-root"
        >
          Location:
        </span>
        <p
          class="MuiTypography-root MuiTypography-body-md css-1ygy07m-JoyTypography-root"
        >
          , 
        </p>
      </div>
      <a
        href="/venues/5213"
      >
        <button
          aria-label="Explore Venue"
          class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-cb19rj-JoyButton-root"
          type="button"
        >
          Explore
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
